services/galley/src/Galley/API/MLS/Welcome.hs

```diff
@@ -118,29 +117,14 @@ sendLocalWelcomes _con _now _rawWelcome _lclients = do
 
 sendRemoteWelcomes ::
   ( Member FederatorAccess r,
+    Member BackendNotificationQueueAccess r,
     Member P.TinyLog r
   ) =>
   ByteString ->
   [Remote (UserId, ClientId)] ->
   Sem r ()
 sendRemoteWelcomes rawWelcome clients = do
   let req = MLSWelcomeRequest . Base64ByteString $ rawWelcome
-      rpc = fedClient @'Galley @"mls-welcome" req
-  traverse_ handleError <=< runFederatedConcurrentlyEither clients $
-    const rpc
-  where
-    handleError ::
-      Member P.TinyLog r =>
-      Either (Remote [a], FederationError) (Remote MLSWelcomeResponse) ->
-      Sem r ()
-    handleError (Right x) = case tUnqualified x of
-      MLSWelcomeSent -> pure ()
-      MLSWelcomeMLSNotEnabled -> logFedError x (errorToWai @'MLSNotEnabled)
-    handleError (Left (r, e)) = logFedError r (toWai e)
-
-    logFedError :: Member P.TinyLog r => Remote x -> Wai.Error -> Sem r ()
-    logFedError r e =
-      P.warn $
-        Logger.msg ("A welcome message could not be delivered to a remote backend" :: ByteString)
-          . Logger.field "remote_domain" (domainText (tDomain r))
-          . logErrorMsg e
+      rpc = void $ fedQueueClient @'Galley @"mls-welcome" req
+  for_ clients $ \client -> do
+    enqueueNotification client Q.Persistent rpc
```

Comment:
Is this really a notification? Sounds like you can send welcome to anyone who you are not even connected to.

Also, there doesn't seem to be a E2E test for this.